server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleQuery.java

```diff
@@ -224,12 +230,23 @@ public RuleQuery setAscendingSort(boolean b) {
     return this;
   }
 
+  @Deprecated
   public RuleQuery setAvailableSince(@Nullable Date d) {
-    this.availableSince = d;
+    this.availableSince = d.getTime();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'd' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)